shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -195,17 +178,11 @@ class JSBackend(allowUnresolvedSymbols: Boolean) {
       JSRecord(fields map { case (key, Fld(_, _, _, value)) =>
         key.name -> translateTerm(value)
       })
-    case Sel(Var(cls), Var(u: "unapply")) =>
+    case Sel(Var(cls), Var(f)) =>
       scope.resolveValue(cls) match {
-        case S(sym: NuTypeSymbol with RuntimeSymbol) =>
-          if (sym.isPlainJSClass) translateNuTypeSymbol(sym).member(u)
-          else translateNuTypeSymbol(sym).member("class").member(u)
-        case S(sym @ CapturedSymbol(out, cls: NewClassSymbol)) =>
-          if (cls.isPlainJSClass)
-            translateCapture(sym).member(u)
-          else
-            translateCapture(sym).member("class").member(u)
-        case _ => JSField(translateTerm(Var(cls)), u)
+        case S(sym: NewClassSymbol) if !sym.isPlainJSClass =>
```

Comment:
Why is this case needed? I don't know why you make a special case here and in `translateVar`:

```scala
      case S(sym: NuTypeSymbol with RuntimeSymbol) =>
        if (sym.isPlainJSClass || !isCallee) translateNuTypeSymbol(sym)
        else translateNuTypeSymbol(sym).member("class")
```

What's the justification? Please provide an example where the behavior we want is not just to generate a plain reference to the symbol.

Reply:
The unnecessary case has been removed.
When we generate code for `a instanceof  ...` or `new ...`, we should use the actual class (i.e., `ClassName.class`) rather than the constructor function. I indicated all the cases where we need `.class` by commenting.